### DISCUSSION TITLE
ISSUE- Skywalking argent error after enabling TLS

### DISCUSSION THREAD

**Discussion by skundu1234** [apache/skywalking#8040] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

After enabling TLS from agent and collector side ,we can see below error. Agent and collector both are using same host. Please let me know if you need any details.
Errror in skywalking-api.log:

INFO 2021-10-29 09:54:03:754 main PluginResourcesResolver : find skywalking plugin define in jar:file:/prometheus/apache-skywalking-apm-bin-es7/agent/activations/apm-toolkit-logback-1.x-activation-8.7.0.jar!/skywalking-plugin.def
ERROR 2021-10-29 09:54:09:418 grpc-default-executor-1 EventReportServiceClient : Failed to report starting event.
org.apache.skywalking.apm.dependencies.io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
Channel Pipeline: [SslHandler#0, ProtocolNegotiators$ClientTlsHandler#0, WriteBufferingAndExceptionHandler#0, DefaultChannelPipeline$TailContext#0]
        at org.apache.skywalking.apm.dependencies.io.grpc.Status.asRuntimeException(Status.java:533)

![image](https://user-images.githubusercontent.com/87635275/139456653-715bf825-9592-43c8-822b-b2a60dbae9e6.png)

Agent:
[esmmon@hostname ca]$ pwd
/prometheus/apache-skywalking-apm-bin-es7/agent/ca
[esmmon@hostname ca]$ ls
ca.crt

agent.config:   I have changed these two lines and added ca.crt as well.
# If true, SkyWalking agent uses TLS even no CA file detected.
 agent.force_tls=${SW_AGENT_FORCE_TLS:true}

# Backend service addresses.
collector.backend_service=${SW_AGENT_COLLECTOR_BACKEND_SERVICES:XX.XX.XX.XX:11800}(XX-IP of collector)

![image](https://user-images.githubusercontent.com/87635275/139457111-f8949da8-0ed9-4c03-9702-9171e69e8110.png)

Application.yml:  I have changed those and did not see any error skywalking.log
core:
  selector: ${SW_CORE:default}

![image](https://user-images.githubusercontent.com/87635275/139457739-cf0e9f16-4d79-409c-a9a0-0ee62e810d06.png)

### What you expected to happen

I expected to work TLS without any issue after doing this change.

### How to reproduce

I have provided all the details and let me know if you need any other details as well to provide.I will provide

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
https://github.com/apache/skywalking/blob/master/test/e2e-v2/cases/simple/mtls/docker-compose.yml

Take a reference from here. We tested mTLS for every commit. And it works.

**Reply by skundu1234** [reply]
Thanks for that.Can you please suggest that configuration for TLS ,gRPC channel for VM.I have changed only core default "core:
selector: ${SW_CORE:default}"  ,should I change that "receiver-sharing-server" option as well. If yes, what should I use for below configuration or any other thing needs to be updated.

receiver-sharing-server:
  selector: ${SW_RECEIVER_SHARING_SERVER:default}
  default:
    # For gRPC server
    gRPCHost: ${SW_RECEIVER_GRPC_HOST:0.0.0.0}
    gRPCPort: ${SW_RECEIVER_GRPC_PORT:0}

**Reply by wu-sheng** [reply]
Activate sharing server if you need TLS, we use core channel for cluster internal communication, which doesn't expect mTLS.

**Reply by skundu1234** [reply]
Thank you.I used now this below but got below error in agent. Can you please suggest if anything needs to be changed.

application.yml:
# For gRPC server
gRPCHost: ${SW_RECEIVER_GRPC_HOST:127.XX.XX.XX}(IP of the server)
gRPCPort: ${SW_RECEIVER_GRPC_PORT:11811
gRPCSslEnabled: ${SW_RECEIVER_GRPC_SSL_ENABLED:true}

I have used agent.config:
agent.force_tls=${SW_AGENT_FORCE_TLS:true}
collector.backend_service=${SW_AGENT_COLLECTOR_BACKEND_SERVICES:XX.XX.XX.XX:11800}(XX-IP of collector)

Still this error is coming from agent log:
ERROR 2021-11-02 02:57:16:092 grpc-default-executor-1 EventReportServiceClient : Failed to report starting event.
org.apache.skywalking.apm.dependencies.io.grpc.StatusRuntimeException: UNAVAILABLE: io exception

Caused by: org.apache.skywalking.apm.dependencies.io.netty.handler.ssl.NotSslRecordException: not an SSL/TLS record: 000012040000000000000300000004000400100000000600002000000004080000000000000f00010000860700000000000000000000000001485454502f3220636c69656e74207072656661636520737472696e67206d697373696e67206f7220636f72727570742e204865782064756d7020666f722072656365697665642062797465733a20313630333031303037393031303030303735303330336636343036643132663638336362633761306465326335313338

**Reply by wu-sheng** [reply]
1. https://skywalking.apache.org/docs/main/latest/en/setup/backend/backend-expose/
2. https://skywalking.apache.org/docs/main/latest/en/setup/backend/grpc-security/

Check official document, which is always a better way to guide.

And `NotSslRecordException` seems not relative to SkyWalking at all.
